vowpalwabbit/example.h

```diff
@@ -147,4 +152,57 @@ typedef std::vector<example*> multi_ex;
 namespace VW
 {
 void return_multiple_example(vw& all, v_array<example*>& examples);
+
+struct restore_prediction
```

Comment:
These look useful! Can they go in a different header?

Reply:
These have the same function as swap_guard.  I will probably remove these if I can replace them with swap guard.